src/jdk.crypto.cryptoki/share/classes/sun/security/pkcs11/wrapper/CK_PBE_PARAMS.java

```diff
@@ -50,15 +50,15 @@
 
 
 /**
- * class CK_PBE_PARAMS provides all of the necessary information required byte
+ * class CK_PBE_PARAMS provides all the necessary information required by
  * the CKM_PBE mechanisms and the CKM_PBA_SHA1_WITH_SHA1_HMAC mechanism.<p>
  * <B>PKCS#11 structure:</B>
  * <PRE>
  * typedef struct CK_PBE_PARAMS {
- *   CK_CHAR_PTR pInitVector;
- *   CK_CHAR_PTR pPassword;
+ *   CK_BYTE_PTR pInitVector;
```

Comment:
Is `CK_PBE_PARAMS` actually already in use? Will these type changes affect anything else?

Reply:
It seems that some of these classes (representing PKCS#11 structures) were added a long time ago, but never tested or used. In this case, we are sure `CK_PBE_PARAMS` had no usages because it didn't have a useful constructor (just Java's default).

We aligned it to be in sync with the C structure in `pkcs11t.h` (untouched by us):
https://github.com/rh-openjdk/jdk/blob/1e26894969d911776f60eaa557d25a2b389060da/src/jdk.crypto.cryptoki/share/native/libj2pkcs11/pkcs11t.h#L1761-L1770

Those Java changes are paired with `jPbeParamToCKPbeParamPtr()` changes in `p11_convert.c`:
  * https://github.com/rh-openjdk/jdk/commit/1e26894969d911776f60eaa557d25a2b389060da#diff-912743d38ae4d55e8dcf806e62ad3fcfbc30beef1a24c6062ce38adc51fb0297L1658-R1698

Finally, `freeCKMechanismPtr()` was adjusted in `p11_util.c`, since `CK_PBE_PARAMS` fields were being leaked:
  * https://github.com/rh-openjdk/jdk/commit/1e26894969d911776f60eaa557d25a2b389060da#diff-74586de2fbd2a496d8532f76c8186f1026799ef473ab0129cabc83701ad0dc58R420-R428